typescript/infra/scripts/deploy.ts

```diff
@@ -61,18 +71,40 @@ async function main() {
     multiProvider.setSharedSigner(signer);
   }
 
+  let contractVerifier;
+  if (buildArtifactPath) {
+    // fetch explorer API keys from GCP
+    const apiKeys = await fetchExplorerApiKeys();
+    // extract build artifact contents
+    const buildArtifact = extractBuildArtifact(buildArtifactPath);
+    // instantiate verifier
+    contractVerifier = new ContractVerifier(
+      multiProvider,
+      apiKeys,
+      buildArtifact,
+      ExplorerLicenseType.MIT,
+    );
+  }
```

Comment:
this flag is only really used when deploying helloworld because that is a different build artifact to the default core one

if you don't pass the artifact it'll fallback to the default contractverifier in the deployers - albeit without api keys

could refactor the deployer args maybe to set the apikeys on a default deployer?